### PR TITLE
Add invalid state.

### PR DESCRIPTION
**Pull request by JimD2L**

### CODE REVIEW

File: d2l-date-picker.html [hunk 1]
```diff
@@ -156,6 +157,11 @@
 				firstDayOfWeek: {
 					type: Number,
 					value: 0
+				},
+				invalid: {
+					type: Boolean,
+					value: false,
+					reflectToAttribute: true
```
Comment (awikkerink) [inline d2l-date-picker.html]: Is this for validation? Would it be possible to set `aria-invalid` within the component instead of outside of it?

Comment (dlockhart) [reply d2l-date-picker.html]: Hmm, yeah like ideally the component itself knows when it's invalid? Unless we're talking about date ranges where each date is valid but the range could be invalid. In that case, there should likely be a new component like a `<d2l-date-range-picker>` that just wraps the two and handles validation.

File: d2l-date-picker.html [hunk 2]
```diff
@@ -91,6 +91,7 @@
 					on-focus="_handleFocus"
 					aria-describedby$="[[_descriptionId]]"
 					aria-labelledby$="[[_labelId]]"
+					aria-invalid$="[[_computedAriaInvalid(invalid)]]"
```
Comment (dlockhart) [inline d2l-date-picker.html]: Could this not just directly bind to the `invalid` property? i.e. `aria-invalid$="[[invalid]]"`

Comment (JimD2L) [reply d2l-date-picker.html]: Patrick insisted it must be the string `'true'`.
https://www.w3.org/TR/wai-aria-1.1/#aria-invalid

File: d2l-date-picker.html (hunk 1 above)
Comment (JimD2L) [reply d2l-date-picker.html]: That is the case I'm trying to support. We have a date range that uses two of these and want to mark it as invalid when there is a negative range selected.

File: d2l-date-picker.html (hunk 2 above)
Comment (dlockhart) [reply d2l-date-picker.html]: Ah OK cool, yeah looks like it's weirdly a string attribute that can be set to several different things.

File: d2l-date-picker.html (hunk 1 above)
Comment (dlockhart) [reply d2l-date-picker.html]: When you're done, could we publish the date-range component so that others can reuse it?

Comment (JimD2L) [reply d2l-date-picker.html]: I'll see what we can do.

**Pull request merged**